File: src/main/kotlin/me/tomasan7/jecnadesktop/ui/component/LessonPeriodView.java
package me.tomasan7.jecnadesktop.ui.component;

import javafx.scene.control.Label;
import javafx.scene.layout.AnchorPane;
import me.tomasan7.jecnaapi.data.LessonPeriod;

public class LessonPeriodView extends AnchorPane
{
	/** The distance of the corner {@link Label labels} from the sides. */
	private static final double EDGE_DISTANCE = 1d;

	private final LessonPeriod lessonPeriod;
	private final int hourIndex;

	public LessonPeriodView (int hourIndex, LessonPeriod lessonPeriod)
	{
		this.lessonPeriod = lessonPeriod;
		this.hourIndex = hourIndex;

		createHourIndexLabel();
		createPeriodLabel();

		getChildren().addAll(createHourIndexLabel(), createPeriodLabel());

		getStyleClass().add("lesson-period-view");
		getStylesheets().add("/ui/component/LessonPeriodView.css");
	}

	private Label createHourIndexLabel ()
	{
		Label hourIndexLabel = new Label(String.valueOf(hourIndex));
		hourIndexLabel.getStyleClass().add("hour-index-label");

		AnchorPane.setTopAnchor(hourIndexLabel, EDGE_DISTANCE);
		AnchorPane.setLeftAnchor(hourIndexLabel, 0d);
		AnchorPane.setRightAnchor(hourIndexLabel, 0d);

		return hourIndexLabel;
	}

	private Label createPeriodLabel ()
	{
		Label periodLabel = new Label(lessonPeriod.toString());
		periodLabel.getStyleClass().add("period-label");

		AnchorPane.setBottomAnchor(periodLabel, EDGE_DISTANCE);
		AnchorPane.setLeftAnchor(periodLabel, 0d);
		AnchorPane.setRightAnchor(periodLabel, 0d);

		return periodLabel;
	}

	public LessonPeriod getLessonPeriod ()
	{
		return lessonPeriod;
	}
}

File: src/main/kotlin/me/tomasan7/jecnadesktop/ui/component/LessonPeriodView.kt
package me.tomasan7.jecnadesktop.ui.component

import javafx.scene.control.Label
import javafx.scene.layout.AnchorPane
import me.tomasan7.jecnaapi.data.LessonPeriod

class LessonPeriodView(private val hourIndex: Int, val lessonPeriod: LessonPeriod) : AnchorPane()
{
    init
    {
        createHourIndexLabel()
        createPeriodLabel()

        children.addAll(createHourIndexLabel(), createPeriodLabel())

        styleClass.add("lesson-period-view")
        stylesheets.add("/ui/component/LessonPeriodView.css")
    }

    private fun createHourIndexLabel(): Label
    {
        val hourIndexLabel = Label(hourIndex.toString())
        hourIndexLabel.styleClass.add("hour-index-label")

        setTopAnchor(hourIndexLabel, EDGE_DISTANCE)
        setLeftAnchor(hourIndexLabel, 0.0)
        setRightAnchor(hourIndexLabel, 0.0)

        return hourIndexLabel
    }

    private fun createPeriodLabel(): Label
    {
        val periodLabel = Label(lessonPeriod.toString())
        periodLabel.styleClass.add("period-label")

        setBottomAnchor(periodLabel, EDGE_DISTANCE)
        setLeftAnchor(periodLabel, 0.0)
        setRightAnchor(periodLabel, 0.0)

        return periodLabel
    }

    companion object
    {
        /** The distance of the corner [labels][Label] from the sides.  */
        private const val EDGE_DISTANCE = 1.0
    }
}